client.go

```diff
@@ -0,0 +1,153 @@
+package support
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/http/httputil"
+	neturl "net/url"
+)
+
+type HijackHttpOptions struct {
```

Comment:
These option are more or less two separate structs: one for setting up the request and one for the streaming. Maybe we should split this into two structs that are inlined into `HijackHttpOptions`?